source/common.ts

```diff
@@ -0,0 +1,479 @@
+    }
+
+    return i18n._normalizeWithAncestorsCache[locale];
+  },
+  _normalizeWithAncestorsCache: {} as Record<string, readonly string[]>,
+  _translations: {} as JSONObject,
+  _ts: 0,
+  __(...args: unknown[]) {
+    // This will be aliased to i18n.getTranslation.
+    return '';
+  },
+  addTranslation(locale: string, ...args: unknown[]) {
+    // This will be aliased to i18n.addTranslations.
+    return {};
+  },
+  addTranslations(locale: string, ...args: unknown[]) {
+    const translation = args.pop() as JSONObject;
+    const path = args.join('.').replace(/(^\.)|(\.\.)|(\.$)/g, '');
+
+    if (typeof translation === 'string') {
+      set(i18n._translations, `${i18n.normalize(locale)}.${path}`, translation);
+    } else if (typeof translation === 'object' && !!translation) {
+      Object.keys(translation)
+        .sort()
+        .forEach(key => {
+          i18n.addTranslations(locale, `${path}.${key}`, translation[key]);
+        });
+    }
+
+    return i18n._translations;
+  },
+  createComponent(
+    translatorSeed?: string | ((...args: unknown[]) => string),
+    locale?: string,
+    reactjs?: typeof import('react'),
+    type?: React.ComponentType | string,
+  ) {
+    const translator = typeof translatorSeed === 'string'
+      ? i18n.createTranslator(translatorSeed, locale)
+      : typeof translatorSeed === 'undefined'
+      ? i18n.createTranslator()
+      : translatorSeed;
+
+    if (!reactjs) {
+      if (typeof React !== 'undefined') {
+        reactjs = React;
+      } else {
+        try {
+          reactjs = require('react');
+        } catch (error) {
+          // Ignore.
+        }
+      }
+
+      if (!reactjs) {
+        console.error('React is not detected!');
+      }
+    }
+
+    type Props = {
+      _containerType?: React.ComponentType | string;
+      _props?: {};
+      _tagType?: React.ComponentType | string;
+      _translateProps?: string[];
+      children?: React.ReactNode;
+    };
+
+    return class T extends reactjs!.Component<Props> {
+      static __ = translator;
+
+      _invalidate = () => this.forceUpdate();
+
+      render() {
+        const {
+          _containerType,
+          _props = {},
+          _tagType,
+          _translateProps,
+          children,
+          ...params
+        } = this.props;
+
+        const tagType = _tagType || type || 'span';
+        const items = reactjs!.Children.map(children, (item, index) => {
+          if (typeof item === 'string' || typeof item === 'number') {
+            return reactjs!.createElement(tagType, {
+              ..._props,
+              dangerouslySetInnerHTML: { __html: translator(item, params) },
+              key: `_${index}`,
+            } as any);
+          }
+
+          if (Array.isArray(_translateProps)) {
+            const newProps: Record<string, string> = {};
+            _translateProps.forEach(propName => {
+              const prop = (item as any).props[propName];
```

Comment:
Why need the cast to `any`?

Reply:
Yes, `item` is technically `ReactNode | ReadonlyArray<ReactNode>` here.